### PR TITLE
Adds verified configuration and testing to issue #14562

### PR DESCRIPTION
**Pull request by carlpulley-da**
* Adds verified configuration and testing to issue #14562

CHANGELOG_BEGIN
    
* DAML Script Runner: DAR upload code in DAML script runner moved to integration tests.
    
CHANGELOG_END

- [x] Refactor `RunnerMain` so that `RunnerConfig` is verified
- [x] Add in unit test to ensure configurations may load and be verified successfully
- [x] Rebase PR when https://github.com/digital-asset/daml/pull/14631 is merged
- [x] Ensure PR CI builds

### CODE REVIEW

File: daml-script/runner/BUILD.bazel [hunk 1]
```diff
@@ -63,14 +63,16 @@ da_scala_library(
 da_scala_test(
     name = "tests",
     srcs = glob(
-        ["src/test/**/*.scala"],
+        ["src/test/scala/**/*.scala"],
```
Comment (carlpulley-da) [inline daml-script/runner/BUILD.bazel:66]: Changes in this file support the use of a `resource` and `scala` directory in `daml-script/src/test`

File: daml-script/runner/src/main/scala/com/digitalasset/daml/lf/engine/script/RunnerMain.scala [hunk 2]
```diff
@@ -22,122 +22,139 @@ import com.daml.lf.iface.reader.InterfaceReader
...
+      if (runnerConfig.jsonApi) {
         Http().shutdownAllConnectionPools().flatMap { case () => system.terminate() }
       } else {
         system.terminate()
       }
     )
     Await.result(flow, Duration.Inf)
   }
+
+  final case class VerifiedRunnerConfig private (
```
Comment (carlpulley-da) [inline daml-script/runner/src/main/scala/com/digitalasset/daml/lf/engine/script/RunnerMain.scala:77]: Using a `private` constructor and a `NoCopy` trait we allow the compiler to control who may create or modify our configuration objects after they have been constructed.

Comment (cocreature) [reply daml-script/runner/src/main/scala/com/digitalasset/daml/lf/engine/script/RunnerMain.scala:77]: I think a bunch of other classes use `CliConfig` & `Config` to distinguish between the one set on the Cli and the validated one. You could consider doing something similar.

**Review by cocreature — APPROVED** [review]
nice cleanup thx!

**Review by remyhaemmerle-da — APPROVED** [review]
Nice.

**Pull request merged**